### ISSUE TITLE
bug when serializing arrays

### ISSUE THREAD

**Issue by friuns2** [mgholam/fastJSON#68]
This code throwing Exception: Cannot determine type
json string it generates:
[{"$types":{"Test":"1"},"$type":"1"},{"$type":"1"}]

```cs
void Main()
{
	string s = fastJSON.JSON.ToJSON(new Test[]{new Test(),new Test()});
	fastJSON.JSON.ToObject<Test[]>(s);
}


public class Test
{
	
}

**Comment by mgholam** [comment]
If `UseExtensions = false` then it works properly, the problem seems to be with the `{"$types"...` at the start.

I will work on it, thanks!

**Comment by mgholam** [comment]
Fixed in 2.1.28
